app/assets/javascripts/admin/addon/components/admin-config-areas/user-fields-list.gjs

```diff
@@ -0,0 +1,90 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+import AdminConfigAreaEmptyList from "admin/components/admin-config-area-empty-list";
+import AdminUserFieldItem from "admin/components/admin-user-field-item";
+import UserField from "admin/models/user-field";
+
+export default class AdminConfigAreasUserFieldsList extends Component {
+  @service dialog;
+  @service store;
+  @service adminUserFields;
+
+  get fieldTypes() {
+    return UserField.fieldTypes();
+  }
+
+  get fields() {
+    return this.adminUserFields.userFields;
+  }
+
+  get sortedFields() {
+    return this.adminUserFields.sortedUserFields;
+  }
+
+  @action
+  moveUp(f) {
+    const idx = this.sortedFields.indexOf(f);
+    if (idx) {
+      const prev = this.sortedFields.objectAt(idx - 1);
+      const prevPos = prev.get("position");
+
+      prev.update({ position: f.get("position") });
+      f.update({ position: prevPos });
+    }
+  }
+
+  @action
+  moveDown(f) {
+    const idx = this.sortedFields.indexOf(f);
+    if (idx > -1) {
+      const next = this.sortedFields.objectAt(idx + 1);
+      const nextPos = next.get("position");
+
+      next.update({ position: f.get("position") });
+      f.update({ position: nextPos });
+    }
+  }
+
+  @action
+  destroyField(f) {
+    this.dialog.yesNoConfirm({
+      message: i18n("admin.user_fields.delete_confirm"),
+      didConfirm: () => {
+        this.#deleteField(f);
+      },
+    });
+  }
+
+  async #deleteField(field) {
+    try {
+      await field.destroyRecord();
+      this.fields.removeObject(field);
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  <template>
+    {{#if this.fields}}
+      {{#each this.sortedFields as |uf|}}
```

Comment:
I also don't very much like 2 letters variable when we can avoid it :D

Reply:
Totally agree! They almost never make things better these 1 or 2 letter vars